### ISSUE TITLE
gradient.addColorStop is not a function

### ISSUE THREAD

**Issue by trickeyd** [iddan/react-native-canvas#25]
Hi, just found this module today, looks great so far. 

However the project I am trialling it for requires radial gradients, but i get the error above when I try to add a color stop to a gradient.

Thanks!

**Comment by trickeyd** [comment]
These are the properties of the object returned by createRadialGradient:

_40: 0
_65: 0
_55: null
_72: null
then: null
catch: null
done: null
finally: null

**Comment by iddan** [comment]
Hey there, thanks for trying the module and reporting.
`react-native-canvas` returns promises instead of values for all of it's methods so if you're code looks like this:
```javascript
class App extends Component {
    handleCanvas = (canvas) => {
         const ctx = canvas.getContext('2d')
         const gradient = ctx.createRadialGradient(x0, y0, r0, x1, y1, r1)
         gradient.addColorStop(offset, color)
```
It should look like:
```javascript
class App extends Component {
    handleCanvas = async (canvas) => {
         const ctx = canvas.getContext('2d')
         const gradient = await ctx.createRadialGradient(x0, y0, r0, x1, y1, r1)
         gradient.addColorStop(offset, color)
```
By `await`-ing for the gradient object you can access it's methods.
Please share if that worked for you or need more help.

**Comment by trickeyd** [comment]
Thanks for coming back. Yes I saw it was a promise so I tried:

```

handleCanvas = (canvas) => { 
const ctx = canvas.getContext("2d")
const gradient = ctx.createRadialGradient(x0, y0, r0, x1, y1, r1)
         .then(gradient => {
                 gradient.addColorStop(offset, color)
         })

```
But the promise never resolves.

I just tried your implementation too - first time I've used async :-) but it never gets past await. So in the example bellow, "after" is never logged.


```
        console.log('before');
        gradient = await ctx.createRadialGradient(36.3, 39.6, 0.0, 36.3, 39.6, 36.5);
        console.log('after');

```

**Comment by iddan** [comment]
Will look into it

**Comment by trickeyd** [comment]
Hello - I was just wondering if you have had a chance to look into this at all. I'm hoping that it's just to do with my own implementation.

Have you tried to use the method yourself?

**Comment by trickeyd** [comment]
(Thanks BTW - I'm sure you're busy)

**Comment by iddan** [comment]
I will try it later today

**Comment by trickeyd** [comment]
Hey - I updater to:  "react-native-canvas": "^0.1.9"

Unfortunately it still doesn't work I'm afraid.

The Promise now resolves but it appears to return an empty object.

`console.log('before');
        gradient = await ctx.createRadialGradient(36.3, 39.6, 0.0, 36.3, 39.6, 36.5);
        console.log('after', gradient);

        // before
        // after Object {}
`

**Comment by iddan** [comment]
gradient should be an empty object

**Comment by iddan** [comment]
Is it renders well?

**Comment by trickeyd** [comment]
It has no methods :-(

On 5 Sep 2017 10:52 pm, "Iddan Ahahronson" <notifications@github.com> wrote:

Is it renders well?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/iddan/react-native-canvas/issues/25#issuecomment-327200747>,
or mute the thread

**Comment by iddan** [comment]
Oh, will look into it

**Comment by iddan** [comment]
resolved in version 0.1.18

**Comment by akiladevinda** [comment]
Still Having this issue. After set 'await' nothing will return

This is my code

```
async function resetCanvas(canv) {
  savedPoints = [];
  pointIndex = 1;
  toolState = 'none';
  linePath = [];
  annotations = [];

  pointIndex = 1;
  //axisX      = margin;
  //axisY      = titleHeight + margin;
  barWidth = 20;
  barY = axisY;
  axisHeight = height - titleHeight - subtitleHeight - 2 * margin;
  barHeight = axisHeight;
  axisWidth = width - 2 * margin - barWidth - 70;
  barX = axisX + axisWidth + margin;
  colors = [];
  for (let i = rgbGrd.length - 1; i >= 0; i--) {
    colors.push(
      'rgb(' + rgbGrd[i][0] + ',' + rgbGrd[i][1] + ',' + rgbGrd[i][2] + ')',
    );
  }
  grd = await canv.createLinearGradient(0, barY, 0, barY + barHeight);
  for (let i = 0; i < colors.length; i++) {
    grd.addColorStop(i / (colors.length - 1), colors[i]);
  }
  // document.getElementById("point").innerHTML =
  //   "Move mouse over scan to see height values.<br>Click to save a point in the form: (ID), Y, X, Z";
  canv.clearRect(0, 0, width, height);
  canv.fillStyle = 'white';
  canv.fillRect(0, 0, width, height);
  //canv.stroke();
  canv.beginPath();
  //   enableButtons(true);

  //canv.clearRect(0, 0, width, height);
  //canv.fillStyle="white";
  //canv.fillRect(0, 0, width-1, height-1);
  //canv.fillStyle="black";
  //canv.rect(0, 0, width, height);
  //canv.beginPath();
}
```
